.github/workflows/consumer_contract_tests.yml

```diff
@@ -90,9 +95,39 @@ jobs:
           echo "repo-branch=${{ steps.extract-branch.outputs.repo-branch }}"
           echo "repo-version=${{ steps.extract-branch.outputs.repo-version }}"
 
+  leo-build-tag-publish-job:
```

Comment:
We might want to break up the build into a reusable workflow or action.

I'm working on finishing up your old https://github.com/DataBiosphere/leonardo/pull/2987/files but we probably want to be able to avoid double building per PR.